### PR TITLE
Multivariate polynomials

### PR DESCRIPTION
**Pull request by xrchz**

### CODE REVIEW

File: examples/algebra/multipoly/multipolyScript.sml [hunk 1]
```diff
@@ -0,0 +1,2232 @@
...
+  \\ pop_assum SUBST1_TAC
+  \\ `(λP. GBAG g (BAG_FILTER P b)) = λP. GBAG g (BAG_IMAGE I (BAG_FILTER P b))`
+  by simp[FUN_EQ_THM]
+  \\ pop_assum SUBST1_TAC
+  \\ irule GBAG_IMAGE_PARTITION
+  \\ simp[]
+  \\ metis_tac[]
+QED
+
+Theorem ring_mult_lsum:
```
Comment (mn200) [inline examples/algebra/multipoly/multipolyScript.sml:417]: Indentation here is gross.

**Comment by mn200** [comment]
Does this subsume #936 ?

**Comment by xrchz** [comment]
> Does this subsume #936 ?

It builds on top of it. So yes, in a sense.

File: examples/algebra/multipoly/multipolyScript.sml (hunk 1 above)
Comment (xrchz) [reply examples/algebra/multipoly/multipolyScript.sml:417]: I don't see the issue - what's the proposed change?

Comment (xrchz) [reply examples/algebra/multipoly/multipolyScript.sml:417]: oh maybe GitHub was showing me the wrong quote - do you mean the statement of ring_mult_lsum? I will improve that.

**Comment by mn200** [comment]
At some point, you will also need to put this into the build sequence.

**Comment by xrchz** [comment]
`examples/algebra` is already in `tools/sequences/final-examples` -- is that not sufficient? I don't know if those sequences imply recursion into subdirectories (if not, then the current entry doesn't accomplish much).

**Comment by mn200** [comment]
The sequences don't imply recursion, but the `Holmakefile` in `examples/algebra` achieves the recursion by mentioning the relevant directories in its `INCLUDES =` line.

**Pull request merged**